setcontext/setcontext.py

```diff
@@ -0,0 +1,406 @@
+
+import fire
+import functools
+import json
+import os
+import re
+import pathlib
+import subprocess
+import sys
+from termcolor import cprint
+
+
+#Add pytest
+#Build framework for  commandline tools.
+
+"""
+Global variable for local git repo.
+"""
+DEBUG = True
+PATH = pathlib.Path.home() / "git"
+
+
+class CONTEXT:
+    """
+
+    Enum type class to remove strings from code.
+    These are the essential environment variable names.
+
+    """
+    PROJECT = "PROJECT"
+    SERVICE = "SERVICE"
+    VERSION = "VERSION"
+
+
+def is_project_name_valid_for_gcloud(name: str) -> bool:
+    """
+
+    Tests for valid name for a gcloud project name.
+
+    :param name:
+        Project name.
+    :return:
+        True if name only has lower space letters, digits and a dash
+    """
+
+    patterns = '^[a-z0-9/-]*$'
+    if re.search(patterns, name):
+        return True
+    else:
+        return False
+
+@functools.lru_cache
+def does_gcloud_project_exist(project_name: str) -> bool:
+    """
+
+    Test to see if a gcloud project exists.
+
+    :param project_name:
+        Name of project.
+    :return: bool
+        True if a gcloud project exists with the name of project_name
+    """
+
+    output = subprocess.check_output("gcloud projects list --format json".split(" "))
```

Comment:
Python3 recommended way to use subprocess is
`subprocess.run(cmd, capture_output=True)`
https://docs.python.org/3/library/subprocess.html#subprocess.run